packaging/rpm-ostree.spec

```diff
@@ -132,12 +138,11 @@ Requires:       librepo%{?_isa} >= %{librepo_version}
 # rpm-ostree wraps more of ostree (such as `ostree admin unlock` etc.)
 Requires: ostree
 Requires: bubblewrap
-Requires: fuse
+Requires: fuse3
```

Comment:
This seems to come from https://src.fedoraproject.org/rpms/rpm-ostree/c/3c602a23787fd2df873c0b18df3133c9fec4b66a, but the reasoning there is not sufficient.

OSTree already lists its fuse dependencies as needed. git archeology shows the fuse here is because we call `fusermount -u`: https://github.com/coreos/rpm-ostree/issues/443.

`fuse3` ships `fusermount3` and I'm not sure if they're supposed to be interchangeable for unmount purposes (though I would assume it mostly boils down to a `umount()` syscall) but clearly using `fusermount -u` to unmount a FUSE mount from rofiles-fuse (compiled with libfuse3) has been working for a while, ever since we started compiling ostree in Fedora against libfuse3.

Hmm, wonder if this is the root cause of https://github.com/coreos/coreos-assembler/issues/3848 somehow; maybe that mismatch stopped working recently?

Anyway, I think ideally there'd be an `rofiles-fuse -u` which would just proxy to the right `fusermount` vs `fusermount3` based on compiled options. Barring that, in practice I think this is fine, but we should probably change our code to use `fusermount3 -u` also.

Reply:
> Hmm, wonder if this is the root cause of https://github.com/coreos/coreos-assembler/issues/3848 somehow; maybe that mismatch stopped working recently?

At least testing locally with latest fuse bits, it seems to work fine to unmount with `fusermount -u` an rofiles-fuse mount, so maybe not.

But still, we need the Requires here to agree with the actual binary we use in the code.

Reply:
maybe we can build a f40 rpm with `fuse` instead of `fuse3` and see if the issue is still around?

Reply:
cc also https://github.com/fedora-silverblue/issue-tracker/issues/589

Reply:
Yeah, definitely a mess here...a perfect example of the pitfalls of trying to maintain spec files upstream.

A larger fix of course is for us to just stop using rofiles-fuse (and go with overlayfs, but...)

Reply:
Won't this break if we explicitly call `fusermount` as the `fuse3` package does not provide such a binary?

We should likely revert that change until we've moved the logic to fuse3 or the wrapper.

Reply:
> Won't this break if we explicitly call `fusermount` as the `fuse3` package does not provide such a binary?

Yup, this is what I meant higher up with:

> we need the Requires here to agree with the actual binary we use in the code.

I think rather than reverting, it makes more sense to just plow forward and move over to the v3 version. Pushed a commit for that.

Reply:
> Yeah, definitely a mess here...a perfect example of the pitfalls of trying to maintain spec files upstream.

Yeah, I'd be OK revisiting that. Historically, the canonical version lived here because it often required changes along with upstream changes (e.g. bundled libdnf deps). Since now things are calmer, it might make sense to just move it back to dist-git.